jgrapht-core/src/test/java/org/jgrapht/graph/SerializationTest.java

```diff
@@ -17,77 +17,660 @@
+ * <p>
+ * The following classes are tested here:
+ * <ul>
+ * <li>{@link SimpleGraph}</li>
+ * <li>{@link Multigraph}</li>
+ * <li>{@link Pseudograph}</li>
+ * <li>{@link DefaultUndirectedGraph}</li>
+ *
+ * <li>{@link SimpleWeightedGraph}</li>
+ * <li>{@link WeightedMultigraph}</li>
+ * <li>{@link WeightedPseudograph}</li>
+ * <li>{@link DefaultUndirectedWeightedGraph}</li>
+ *
+ * <li>{@link SimpleDirectedGraph}</li>
+ * <li>{@link DirectedMultigraph}</li>
+ * <li>{@link DirectedPseudograph}</li>
+ * <li>{@link DefaultDirectedGraph}</li>
+ *
+ * <li>{@link SimpleDirectedWeightedGraph}</li>
+ * <li>{@link DirectedWeightedMultigraph}</li>
+ * <li>{@link DirectedWeightedPseudograph}</li>
+ * <li>{@link DefaultDirectedWeightedGraph}</li>
+ * </ul>
  *
  * @author John V. Sichi
  */
 public class SerializationTest
 {
-    // ~ Instance fields --------------------------------------------------------
+    private static final String v1 = "v1";
+    private static final String v2 = "v2";
+    private static final String v3 = "v3";
+    private static final List<String> vertexList = Arrays.asList(v1, v2, v3);
+    private static final List<List<String>> vertexPairs = Arrays
+        .asList(Arrays.asList(v1, v2), Arrays.asList(v2, v1), Arrays.asList(v1, v3),
+            Arrays.asList(v3, v1), Arrays.asList(v2, v3), Arrays.asList(v3, v2));
+
+    public static <V, E> void assertContainsAllVertices(Graph<V, E> graph, List<V> vertices)
+    {
+        for (V v : vertices) {
+            assertTrue(graph.containsVertex(v));
+        }
+    }
+
+    public static <V, E> void checkEdgesOf(Graph<V, E> graph, List<Integer> edges, List<V> vertices)
+    {
+        if (edges.size() != vertices.size()) {
+            throw new IllegalArgumentException(
+                "the size of list of #edges and vertices should match");
+        }
+        for (int i = 0; i < edges.size(); i++) {
+            assertEquals(edges.get(i).intValue(), graph.edgesOf(vertices.get(i)).size());
+        }
+    }
+
+    public static <E> void assertAllEdges(Graph<String, E> graph1, Graph<String, E> graph2)
+    {
+        for (int i = 0; i < vertexPairs.size(); i++) {
+            String a = vertexPairs.get(i).get(0);
+            String b = vertexPairs.get(i).get(1);
+            assertEquals(graph1.getAllEdges(a, b).size(), graph2.getAllEdges(a, b).size());
+            assertEquals(graph1.containsEdge(a, b), graph2.containsEdge(a, b));
+        }
+    }
+
+    private static <E> void verifyBasic(
+        Graph<String, E> graph1, Graph<String, E> graph2, List<Integer> numberOfEdges)
+    {
+        assertContainsAllVertices(graph2, vertexList);
+        assertContainsAllVertices(graph1, vertexList);
+
+        assertAllEdges(graph1, graph2);
+
+        checkEdgesOf(graph2, numberOfEdges, vertexList);
+        checkEdgesOf(graph1, numberOfEdges, vertexList);
+
+        assertEquals(graph1.toString(), graph2.toString());
+    }
+
+    private static <E> void assertWeight(
```

Comment:
Add a method which takes both graph1 and graph2 as inputs here and then apply the checks to both via this method (similar to verifyBasic); that way you can save some redundant code in the invocations.

Reply:
Thanks for the suggestion. I've made the appropriate changes as suggested by you.